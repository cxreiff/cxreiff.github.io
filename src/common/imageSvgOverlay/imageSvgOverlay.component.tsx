import { FC, ReactElement, SVGProps, useState } from 'react'
import cn from 'classnames'

type ImageSvgOverlayProps = {
    aspect: number,
    source: string,
    alt?: string,
    className?: string,
    children: ReactElement<SVGProps<SVGElement>>,
}

import { Frame } from '~/src/common/frame/frame.component'

import styles from './imageSvgOverlay.module.scss'

export const ImageSvgOverlay: FC<ImageSvgOverlayProps> = ({ aspect, source, alt, className, children: svg }) => {

    const [loading, setLoading] = useState(true)

    return (
        <Frame aspect={aspect} loading={loading} className={cn(className, styles.image_svg_overlay)}>
            <img src={source} onLoad={() => setLoading(false)} alt={alt} />
            {!loading && svg}
        </Frame>
    )
}
