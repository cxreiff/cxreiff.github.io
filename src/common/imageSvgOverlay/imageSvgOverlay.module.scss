@use '../../shared.scss' as *;

.image_svg_overlay {
    position: relative;

    .svg_overlay {
        position: absolute;
        inset: 0;
        z-index: 1;
        height: 100%;
        width: 100%;
    }
}
