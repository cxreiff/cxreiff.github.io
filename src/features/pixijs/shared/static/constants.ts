export class Constants {
    public static readonly TEXT_OPTIONS = {
        fontFamily : 'monospace',
        fill : 0x3D3333,
        fontSize: 32,
        align: 'left',
        dropShadow: true,
        dropShadowDistance: 0,
        dropShadowBlur: 3,
        dropShadowColor: 0xEEDDDD,
    }
}
