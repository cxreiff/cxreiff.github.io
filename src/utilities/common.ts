
export const isMobile = () => window.matchMedia(
    "only screen and (hover: none) and (max-width: 768px)"
).matches
