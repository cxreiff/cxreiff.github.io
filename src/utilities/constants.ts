export const OBJECT_STORE_ASSETS_URL = 'https://jaxreiff-static.sfo3.cdn.digitaloceanspaces.com/'