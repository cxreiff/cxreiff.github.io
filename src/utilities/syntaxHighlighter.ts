import { CSSProperties } from 'react'
import { PrismLight } from 'react-syntax-highlighter'
import { prism } from 'react-syntax-highlighter/dist/esm/styles/prism'

/**
* Import and register languages for highlighting as-needed.
* (Importing individually for the purpose of keeping bundle size low)
*/

import tsx from 'react-syntax-highlighter/dist/esm/languages/prism/tsx'
PrismLight.registerLanguage('tsx', tsx)

import scss from 'react-syntax-highlighter/dist/esm/languages/prism/scss'
PrismLight.registerLanguage('json', scss)

export const SyntaxHighlighter = PrismLight
export const syntaxStyle = prism as CSSProperties
