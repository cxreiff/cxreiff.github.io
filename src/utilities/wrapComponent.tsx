import React, { ComponentType, ReactElement } from 'react';
import { Provider } from 'react-redux';
import { ConnectedRouter } from 'connected-react-router';

import store from '~/src/app/store';
import history from '~/src/app/history';

const wrapComponent = (Component: ComponentType, props = {}): ReactElement => {
  return (
    <Provider store={store}>
      <ConnectedRouter history={history}>
        <Component {...props} />
      </ConnectedRouter>
    </Provider>
  )
}

export default wrapComponent
