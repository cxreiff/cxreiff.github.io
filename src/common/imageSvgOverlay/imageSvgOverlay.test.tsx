import { test, expect } from "vitest";
import { render } from "@testing-library/preact";

import { ImageSvgOverlay } from "./imageSvgOverlay.component";
import { IMAGE_URLS } from "src/utilities/constants";

test("should render", () => {
  const { container } = render(
    <ImageSvgOverlay aspect={3 / 2} source={IMAGE_URLS.ABOUT_ME}>
      <svg />
    </ImageSvgOverlay>
  );

  expect(container).toMatchSnapshot();
});
