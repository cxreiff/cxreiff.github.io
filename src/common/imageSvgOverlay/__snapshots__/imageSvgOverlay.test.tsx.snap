// Vitest Snapshot v1

exports[`should render 1`] = `
<div>
  <div
    class="_frame_331539 _image_svg_overlay_003598 _hidden_331539"
  >
    <img
      src="https://cxreiff.sfo3.cdn.digitaloceanspaces.com/assets/aboutme.png"
    />
    <svg
      viewBox="0 0 1.5 1"
    />
  </div>
</div>
`;
